torchvision/models/swin_transformer.py

```diff
@@ -335,19 +540,26 @@ def __init__(
         attention_dropout: float = 0.0,
         stochastic_depth_prob: float = 0.0,
         num_classes: int = 1000,
-        norm_layer: Optional[Callable[..., nn.Module]] = None,
         block: Optional[Callable[..., nn.Module]] = None,
+        norm_layer: Optional[Callable[..., nn.Module]] = None,
+        use_v2: bool = False,
```

Comment:
I think we can avoid using `use_v2` here.
Not sure if this is the best approach, but here is my idea:
1) We can let `block` by default `SwinTransformerBlock` as before. And for `v2` builder we should explicitly pass the `block=SwinTransformerBlockv2`.
2) For `v2_pretrained_window_sizes` I think we can just have the if - else logic based on whether the value is `None`
3) I think we should let `downsample_layer` as a parameters now where the default is `PatchMerging` but you can pass `PatchMergingV2` on the `swin_v2` builder

Let me know what are your thought on this :)